### DISCUSSION TITLE
Can't find a ministry inside a ministry office building

### DISCUSSION THREAD

**Discussion by mtmail** [osm-search/Nominatim#3387] [Geocoding around the world]
In Pristina, Kosovo there's a building tagged `office=government`, `government=ministry`. https://www.openstreetmap.org/way/125541064 On the OSM map style that's rendered as boundary plus a node.

https://wiki.openstreetmap.org/wiki/Tag:government=ministry

Inside the building there are two more ministries:
- south: Ministry of Agriculture, Forestry and Rural Development https://www.openstreetmap.org/node/1421041089
- north: Ministry of Culture, Youth and Sport https://www.openstreetmap.org/node/1421032344

With Nominatim 4.4. no matter where I click on the reverse search always the building is returned, never one of the nodes. I tried setting `layer=poi` and `layer=address`, too. In Nominatim 4.3 it works though.

https://nominatim.openstreetmap.org/ui/reverse.html?lat=42.66115&lon=21.16226&zoom=18

![image](https://github.com/osm-search/Nominatim/assets/3727288/d6737ab6-fb5d-4c74-973d-aa868e8971bb)

**Comment by lonvia** [comment]
We had already discussed this in #2785. I've closed it as a "won't fix" then. And I still don't see a good way to resolve it. However, thinking about it, it seems that the current behaviour prevents from reverse geocoding any amenity within, for example, an airport. That's not very satisfying. Maybe it's time to reopen #2785.
